app_flutter/test/status_grid_test.dart

```diff
@@ -158,5 +161,134 @@ void main() {
+
+      // Snackbar message should go away after its duration
+      await tester.pumpAndSettle(
+          StatusGridContainer.errorSnackbarDuration); // wait the duration
+      await tester.pump(); // schedule animation
+      await tester.pump(const Duration(milliseconds: 1500)); // close animation
+
+      /// Update state to no longer have errors
+      buildState.notifyListeners();
+
+      // Wait another snackbar duration to prevent a race condition and ensure it clears
+      await tester.pumpAndSettle(StatusGridContainer.errorSnackbarDuration);
+
+      expect(
+          find.text(StatusGridContainer.errorFetchCommitStatus), findsNothing);
+      expect(find.text(StatusGridContainer.errorFetchTreeStatus), findsNothing);
+    });
+
+    testWidgets('error snackbar shown for error fetching commit statuses',
+        (WidgetTester tester) async {
+      final FlutterBuildState buildState = TestFlutterBuildState(
+        isTreeBuildingValue: CocoonResponse<bool>()..data = true,
+        statusesValue: CocoonResponse<List<CommitStatus>>()
+          ..data = <CommitStatus>[CommitStatus()]
+          ..error = 'statuses error',
+      );
+
+      await tester.pumpWidget(MaterialApp(
+        home: Scaffold(
+          body: Column(
+            children: <Widget>[
+              ChangeNotifierProvider<FlutterBuildState>.value(
+                value: buildState,
+                child: const StatusGridContainer(),
+              ),
+            ],
+          ),
+        ),
+      ));
+
+      await tester.pump(
+          const Duration(milliseconds: 750)); // open animation for snackbar
+
+      expect(find.text(StatusGridContainer.errorFetchTreeStatus), findsNothing);
+      expect(find.text(StatusGridContainer.errorFetchCommitStatus),
+          findsOneWidget);
+
+      // Snackbar message should go away after its duration
+      await tester.pumpAndSettle(
+          StatusGridContainer.errorSnackbarDuration); // wait the duration
+      await tester.pump(); // schedule animation
+      await tester.pump(const Duration(milliseconds: 1500)); // close animation
+
+      /// Update state to no longer have errors
+      buildState.notifyListeners();
+
+      // Wait another snackbar duration to prevent a race condition and ensure it clears
+      await tester.pumpAndSettle(StatusGridContainer.errorSnackbarDuration);
+
+      expect(
+          find.text(StatusGridContainer.errorFetchCommitStatus), findsNothing);
+      expect(find.text(StatusGridContainer.errorFetchTreeStatus), findsNothing);
+    });
   });
 }
+
+/// A [ChangeNotifier] for testing the interactions with error handling.
+class TestFlutterBuildState extends ChangeNotifier
+    implements FlutterBuildState {
+  TestFlutterBuildState(
+      {@required this.isTreeBuildingValue, @required this.statusesValue});
+
+  // Only return an error on the first call
+  int hasErrorCallCount = 1;
+
+  @override
+  bool get hasError => hasErrorCallCount++ == 1;
```

Comment:
This isn't flexible and straightforward, how about adding a method to set `_error` to true/false, and `hasError` simply returns that value.

Reply:
Added. Switched type to `TestFlutterBuildState` since the type does not matter when being passed in, this allowed manually setting values.